### PR TITLE
Fifo animal shelter

### PR DESCRIPTION
**Pull request by JamesDansie**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed - Kind of, but mostly broken
 - [ ] Unit tests written and passing
     - [ ] “Happy Path” - Expected outcome
     - [ ] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401challenges/src/test/java/code401challenges/utilities/AnimalShelterTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        test = new AnimalShelter();
+        test.enqueue(new Dog("Dog 1"));
+        test.enqueue(new Dog("Dog 2"));
+        test.enqueue(new Cat("Cat 1"));
+        test.enqueue(new Cat("Cat 2"));
+;    }
+
+    @Test
+    public void dequeue() {
+        Animal dog = test.dequeue("dog");
```
Comment (ncarignan) [inline code401challenges/src/test/java/code401challenges/utilities/AnimalShelterTest.java]: Saving your "dog" instance into an `Animal` typed variable makes it so that you no longer can access it's `name` variable because `Animal` is an interface so it has no instance variables. You can change this line to `Dog dog = test.dequeue("dog");`

File: code401challenges/src/main/java/code401challenges/utilities/Animal.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package code401challenges.utilities;
+
+public interface Animal {
+    public void sound();
```
Comment (ncarignan) [inline code401challenges/src/main/java/code401challenges/utilities/Animal.java]: Another fix for your test would be to add a getName method to the animal interface, then you can test both `Cat` and `Dog` instances as type `Animal`
